yellowbrick/features/pcoords.py

```diff
@@ -169,6 +200,37 @@ def draw(self, X, y, **kwargs):
         Called from the fit method, this method creates the parallel
         coordinates canvas and draws each instance and vertical lines on it.
         """
+        # Convert from dataframe
+        if is_dataframe(X):
+            X = X.as_matrix()
+
+        # Choose a subset of samples
```

Comment:
I kind of like having it in `draw` because this preprocessing is a required prerequisite to the actual drawing code but is not specific to either `fit` or `transform`.  So, e.g., if later on someone wanted to move the call to `draw` from `fit` to `transform` (say, in a derived class) they could do so without having to reproduce all the preprocessing by hand.  It also means that here I don't have to worry about which of those methods in the base class calls `draw`, meaning someone could change the base class and no updates would be needed here.  

So, in my opinion it is more elegant this way, but all that being said I'm open to doing it the other way too if you'd prefer.